### PR TITLE
Disallow all plane formulations with new lagrangian kernels in SM physics

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #27340

until it s implemented in the physics action, it will fix the bad error we get now

### CODE REVIEW

File: modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C [hunk 1]
```diff
@@ -533,6 +530,8 @@ QuasiStaticSolidMechanicsPhysics::actSubdomainChecks()
       else if (_out_of_plane_direction != OutOfPlaneDirection::z && _ndisp != 3)
         mooseError("Must specify three displacements for plane strain when the out of plane "
                    "direction is x or y");
+      if (_lagrangian_kernels)
+        paramError("Plane formulation not available with Lagrangian kernels");
```
Comment (GiudGiud) [inline modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C:534]: umm actually wrong location

**Comment by hugary1995** [comment]
Can we use that new `mooseDocumentedError` API (or whatever that is)?

You could extend it to `paramDocumentedError` while you are at it. :)

**Comment by GiudGiud** [comment]
> You could extend it to paramDocumentedError while you are at it. :)

will have to be for another day. I m back from travel and catching up on a lot of stuff

File: modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C [hunk 2]
```diff
@@ -207,6 +207,9 @@ QuasiStaticSolidMechanicsPhysics::QuasiStaticSolidMechanicsPhysics(const InputPa
   // plane strain consistency check
   if (_planar_formulation != PlanarFormulation::None)
   {
+    if (_lagrangian_kernels)
+      mooseDocumentedError(
+          "moose", 27340, "Plane formulation not available with Lagrangian kernels");
```
Comment (hugary1995) [inline modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C:212]:
```suggestion
          "moose", 27340, "Planar formulations are not yet available through the Physics syntax with new_system = true. They can be enabled by manually setting up the appropriate objects. Please refer to the documentation and regression tests for examples.");
```

**Review by hugary1995 — CHANGES_REQUESTED** [review]
I'm guilty of this. Let's hold this off till next Monday. I may be able to _actually_ resolve this issue over the weekend. Let's merge this PR otherwise.

**Review by hugary1995 — APPROVED** [review]
I am working on rewriting the action. But this is definitely a good error message in the interim. Thanks.

**Pull request merged**